zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -782,6 +782,10 @@ void queueCallback(AsyncCallback cb, int rc, String path, Object ctx) {
         eventThread.queueCallback(cb, rc, path, ctx);
     }
 
+    protected void onConnecting(InetSocketAddress addr) {
+        LOG.info("Connecting server {}.", addr);
+    }
+
```

Comment:
- I wonder whether a log is needed here? because I found the `logStartConnect` in the `SendThread#startConnect`?
- If needed, `LOG.info("Start to reconnect to server {}.", addr); `or `LOG.info("Try to reconnect to server {}.", addr)`; is better?

Reply:
Actually onConnecting() method is added to facilitate the test case. onConnecting() is the unsafe point that is described in JIRA issue. We can override this method to inject some actions in test case. The injected actions make test case happen. If we don't have this method, it is difficult to construct a test case to reproduce the issue.
Your suggestion of log changing is good for me. Accept it. I will change the log to "Try to reconnect to server {}." Thanks for your comments.